src/transfer-manager.ts

```diff
@@ -0,0 +1,296 @@
+   * ```
+   * @experimental
+   */
+  async uploadManyFiles(
+    filePaths: string[],
+    options: UploadManyFilesOptions = {}
+  ): Promise<UploadResponse[]> {
+    if (options.skipIfExists && options.passthroughOptions?.preconditionOpts) {
+      options.passthroughOptions.preconditionOpts.ifGenerationMatch = 0;
+    } else if (
+      options.skipIfExists &&
+      options.passthroughOptions === undefined
+    ) {
+      options.passthroughOptions = {
+        preconditionOpts: {
+          ifGenerationMatch: 0,
+        },
+      };
+    }
+
+    const limit = pLimit(
+      options.concurrencyLimit || DEFAULT_PARALLEL_UPLOAD_LIMIT
+    );
+    const promises = [];
+
+    for (const filePath of filePaths) {
+      const stat = await fsp.lstat(filePath);
+      if (stat.isDirectory()) {
+        continue;
+      }
+      const passThroughOptionsCopy: UploadOptions = extend(
+        true,
+        {},
+        options.passthroughOptions
+      );
+      passThroughOptionsCopy.destination = filePath;
+      if (options.prefix) {
+        passThroughOptionsCopy.destination = path.join(
+          options.prefix,
+          passThroughOptionsCopy.destination
+        );
+      }
+      promises.push(
+        limit(() => this.bucket.upload(filePath, passThroughOptionsCopy))
+      );
+    }
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * @typedef {object} DownloadManyFilesOptions
+   * @property {number} [concurrencyLimit] The number of concurrently executing promises
+   * to use when downloading the files.
+   * @property {string} [prefix] A prefix to append to all of the downloaded files.
+   * @property {string} [stripPrefix] A prefix to remove from all of the downloaded files.
+   * @property {object} [passthroughOptions] {@link DownloadOptions} Options to be passed through
+   * to each individual download operation.
+   * @experimental
+   */
+  /**
+   * Download multiple files in parallel to the local filesystem. This is a convenience method
+   * that utilizes {@link File#download} to perform the download.
+   *
+   * @param {array} [files] An array of file objects to be downloaded.
+   * @param {DownloadManyFilesOptions} [options] Configuration options.
+   * @returns {Promise<DownloadResponse[]>}
+   *
+   * @example
+   * ```
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('my-bucket');
+   * const transferManager = new TransferManager(bucket);
+   *
+   * //-
+   * // Download multiple files in parallel.
+   * //-
+   * const response = await transferManager.downloadManyFiles(bucket.File('file1.txt'), bucket.File('file2.txt')]);
+   * // The following files have been downloaded:
+   * // - "file1.txt" (with the contents from my-bucket.file1.txt)
+   * // - "file2.txt" (with the contents from my-bucket.file2.txt)
+   * ```
+   * @experimental
+   */
+  async downloadManyFiles(
```

Comment:
do other lanauges have this as a list of objects? or is a list of strings also an option?

Reply:
Python does it as a list of names. If desired we could move to that as well. Thoughts?

Reply:
The two use cases are 
1) download "directory"
2) download list of files

Not sure which is more common (question for @domZippilli) but it would be great to support both (or show users how to do both)

Reply:
I've modified the signature to now accept an array of files, an array of strings, or a single string. If a single string is provided it will be treated as a GCS prefix to call `getFiles`. Utilizing this prefix will allow for an entire folder structure to be downloaded.

Reply:
That's probably the right idea. It sounds like an idiomatic variation on what I wrote before I saw @ddelgrosso1's reply.

A directory is probably the most common set of files people upload. But of course you'd want to think about this as a list of files to be extensible. To me, the logical progression is something like "I want to upload a directory" -> "I want to form a glob of file names" -> "I want to upload a list of files". So solving the last problem lets you frame the others as variations on it. For example, make a convenience method that accepts a directory name, and just constructs the list of files to pipe into the actual implementation.